### PR TITLE
Remove `-R` flags from the MacOS crosstool template.

### PR DESCRIPTION
**Pull request by jmillikin-stripe**
Fixes https://github.com/bazelbuild/bazel/issues/4028

### CODE REVIEW

**Comment by bazel-io** [comment]
Can one of the admins verify this patch?

**Comment by jmillikin-stripe** [comment]
Bazel devs: this seems like something that could be caught in a regression test, but I'm not familiar enough with the Bazel test suite to write such a test. Is there an existing example "check this target builds" that I could cargo-cult from?

**Comment by jmillikin-stripe** [comment]
Also: the top of `CROSSTOOL.tpl` says it's auto-generated by some internal Google script. This patch may need to be applied to that script instead.

**Comment by jmillikin-stripe** [comment]
Updated with tests per https://github.com/bazelbuild/bazel/issues/4028#issuecomment-344070966

File: src/main/java/com/google/devtools/build/lib/rules/cpp/CppActionConfigs.java [hunk 1]
```diff
@@ -938,24 +938,26 @@ public static String getCppActionConfigs(
...
-                "      flag: '-R'",
-                "      flag: '.gnu.switches.text.cpp_includes'",
-                "      flag: '-R'",
-                "      flag: '.gnu.switches.text.cl_args'",
-                "      flag: '-R'",
-                "      flag: '.gnu.switches.text.lipo_info'",
-                "      flag: '-R'",
-                "      flag: '.gnu.switches.text.annotation'",
-                "    }",
+                ifTrue(
```
Comment (mhlopko) [inline src/main/java/com/google/devtools/build/lib/rules/cpp/CppActionConfigs.java]: Ah I missed this one, can you pls use ifMac there?

File: src/test/shell/bazel/cpp_darwin_integration_test.sh [hunk 2]
```diff
@@ -80,5 +80,19 @@ EOF
...
+  cat >cpp/osx_binary_strip/BUILD <<EOF
+cc_binary(
+  name = "main",
+  srcs = ["main.cc"],
+)
+EOF
+  cat >cpp/osx_binary_strip/main.cc <<EOF
+int main() { return 0; }
+EOF
+  assert_build //cpp/osx_binary_strip:main.stripped >& $TEST_log || fail "//cpp/osx_binary_strip:main.stripped didn't build"
```
Comment (mhlopko) [inline src/test/shell/bazel/cpp_darwin_integration_test.sh]: Can you pls move the fail onto the next line.

File: src/main/java/com/google/devtools/build/lib/rules/cpp/CppActionConfigs.java (hunk 1 above)
Comment (jmillikin-stripe) [reply src/main/java/com/google/devtools/build/lib/rules/cpp/CppActionConfigs.java]: `ifMac` is the opposite of what we want. These lines should only be added on non-MacOS platforms.

File: src/test/shell/bazel/cpp_darwin_integration_test.sh (hunk 2 above)
Comment (jmillikin-stripe) [reply src/test/shell/bazel/cpp_darwin_integration_test.sh]: This matches the existing style. Note that I think there might be something wrong with the current test code -- https://github.com/bazelbuild/bazel/issues/4085 -- would you like me to update both as part of this change?

File: src/main/java/com/google/devtools/build/lib/rules/cpp/CppActionConfigs.java (hunk 1 above)
Comment (mhlopko) [reply src/main/java/com/google/devtools/build/lib/rules/cpp/CppActionConfigs.java]: Then use ifLinux pls, windows crosstool is not patched by this code.

Comment (jmillikin-stripe) [reply src/main/java/com/google/devtools/build/lib/rules/cpp/CppActionConfigs.java]: I believe this should also be applied to FreeBSD.

File: src/test/shell/bazel/cpp_darwin_integration_test.sh (hunk 2 above)
Comment (mhlopko) [reply src/test/shell/bazel/cpp_darwin_integration_test.sh]: Does it? Line 80 in this file is the desired formatting. Let me investigate #4085 first, we run this test on the CI and I don't think we have this issue there. I'll take a look.

Comment (mhlopko) [reply src/test/shell/bazel/cpp_darwin_integration_test.sh]: Hmm you know what, just fix it pls, the fail shouldn't be there twice even if it worked. I didn't know assert_build also fails.

File: src/main/java/com/google/devtools/build/lib/rules/cpp/CppActionConfigs.java (hunk 1 above)
Comment (mhlopko) [reply src/main/java/com/google/devtools/build/lib/rules/cpp/CppActionConfigs.java]: That is covered by 'linux' cppPlatform :) https://github.com/bazelbuild/bazel/blob/275ae45b1228bdd0f912c4fbd634b29ba4180383/src/main/java/com/google/devtools/build/lib/rules/cpp/CppToolchainInfo.java#L353 I know, this is not very robust :)

Comment (jmillikin-stripe) [reply src/main/java/com/google/devtools/build/lib/rules/cpp/CppActionConfigs.java]: OK, changed to use `ifLinux`. I also added a comment that FreeBSD counts as `LINUX`.

File: src/test/shell/bazel/cpp_darwin_integration_test.sh (hunk 2 above)
Comment (jmillikin-stripe) [reply src/test/shell/bazel/cpp_darwin_integration_test.sh]: Done.

**Comment by jmillikin-stripe** [comment]
Gentle ping. Anything else I can do to get this in? The second half of this week is a big US holiday so I won't be around to answer review comments.

**Comment by mhlopko** [comment]
Thanks for the ping, lgtm! Importing. Thank you!

**Comment by jmillikin-stripe** [comment]
Looks like this didn't make it into 0.8. Can it be cherrypicked into 0.8.1, or do we need to wait for 0.9?

**Comment by mhlopko** [comment]
Polished the test internally, plus there was some discussion on the internal review. Hopefully submitting soon.

**Comment by mhlopko** [comment]
Re cherripick, I'm afraid it's too late for 0.8.*, this will have to wait for 0.9. Sorry for that.

**Pull request closed without merging**